src/components/tree/utils.ts

```diff
@@ -0,0 +1,98 @@
+import { bboxPolygon, booleanIntersects, polygon } from '@turf/turf';
+
+import { FilterActiveFilters } from '@/context/FilterContext/types';
+import { Collection } from '@/typings/stac';
+
+import { TreeCatalog } from './Tree';
+
+export const CATALOG = 'Catalog';
+
+const COLLECTION = 'Collection';
+
+const filterByText = (label: string, filter: string) => {
+  return label.trim().toLowerCase().includes(filter.trim().toLowerCase());
+};
+
+const filterByTemporal = (node: Collection, activeFilters: FilterActiveFilters) => {
+  const { start, end } = activeFilters.temporal;
+  if (start && end) {
```

Comment:
If there is just one date (start) then that would be an open ended filter from then onwards.

Reply:
That is not how it works right now, in fact, if you go to test and:

1. enter an AOI
2. enter only a start date
3. select sentinel

Then you get no items at all, so I say that is a bug in the current system, I'd rather record that as a bug and fix separately, I don't see it as an issue specific to the hierarchical view.

Right now, the only filter that works at the collection level is the free text filter, neither the `temporal` or `spatial` work at the collection level, which feels strange, especially to a naive use like myself, who is still pretty new to the app. Again, I would vote that these be recorded as tickets but fixed in an upcoming sprint and not derail this, as I still expect a lot of comments from clients of whether this is even the right approach.

Happy to discuss.

Reply:
Okay, we should consider these changes soon as the `/collections` endpoint allows for it.

Reply:
I'll create a ticket and get it in the backlog